src/testing/run.ts

```diff
@@ -159,14 +159,17 @@ async function runTestCaseUnsafe<TestCaseType, OutputType>(args: {
     throw new Error(`[${args.testId}] Test case semaphore not found.`);
   }
 
-  const output = await testCaseRunAsyncLocalStorage.run(
+  const { output, durationMs } = await testCaseRunAsyncLocalStorage.run(
     {
       testCaseHash: args.testCaseHash,
       testId: args.testId,
     },
     async () => {
       return await semaphore.run(async () => {
-        return await args.fn({ testCase: args.testCase });
+        const startTime = Date.now();
```

Comment:
I think we should use performance.now(). If someone mocks the date api in a test this would break as is. Performance.now is also designed for measuring time between 2 points